src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  createParamDecorator,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+
+import { UserEntity, UsersRepository } from './users.repository';
+import { AUTH_SECRET_KEY } from './auth.module';
+
+export interface JwtPayload {
+  username: string;
+}
+export interface JwtToken {
+  accessToken: string;
+}
+
+export const GetUser = createParamDecorator(
+  (_data, ctx: ExecutionContext): UserEntity => {
+    const req = ctx.switchToHttp().getRequest();
+    return req.user;
+  },
+);
```

Comment:
decorators/guards/interfaces/enums should be in different files, in this case GetUser decorator will be used in the whole app and it will be easier to find when it would have it's own file